File: 2022/01/01.py
def preprocessing(input):
    return {(int(calories) for calories in reeinder.splitlines()) for reeinder in input.split("\n\n")}

def solver(input): 
    calories = sorted(sum(reeinder) for reeinder in input)
    yield calories[-1]
    yield sum(calories[-3:])

File: tmpevents/2022/01/01.rs
aoc::main!();

fn preprocessing(input: &str) -> Vec<Vec<u32>>{
    input
        .split("\n\n")
        .map(|r| r.lines().map(|cal| cal.parse().unwrap()).collect())
        .collect()
}

fn part_1(input: Vec<Vec<u32>>) -> u32 {
    solver(input, 1)
}

fn part_2(input: Vec<Vec<u32>>) -> u32 {
    solver(input, 3)   
}

fn solver(input: Vec<Vec<u32>>, n: usize) -> u32 {
    input
    .iter()
    .map(|r| r.iter().sum::<u32>())
    .sorted_by(|a,b| b.cmp(a))
    .take(n)
    .sum() 
}